ericaltendorf/plotman
# New configuration option for interactive

## What does this change accomplish?
Add a new configuration option to `interactive` and a command line argument to enable or disable automatic plotting when opening the UI. 

Prior to that, anytime `plotman interactive` ran it would automatically fire any pending jobs, even before the user had the chance to press `p` to change its status. 

This PR introduces two new options:
1. A new section in `plotman.yaml` 
2. A command line argument

The section is `tools/interactive/autostart_plotting`, which defaults to True in order to keep backwards compatibility. The command line argument is `--autostart-plotting` and `--no-autostart-plotting`, which take precedence over `plotman.yaml``

## How to test it
1. Set `tools/interactive/autostart_plotting` to `False` and later to `True` and check the `interactive ` reflects the plotting status
2. Pass the command line arguments and check if they override what was set on the configuration file
3. Remove `tools` from the configuration file and run `plotman interactive`. It should show plotting as active (the original behaviour)
